### ISSUE TITLE
Can't Intersect Type.Object() with Type.Record using TemplateLiterals as keys

### ISSUE THREAD

**Issue by cwolter97** [sinclairzx81/typebox#604]
see example
✅ typescript
```typescript
const TempLiteral = Type.TemplateLiteral('foo${number}');
type TempLiteralType = Static<typeof TempLiteral>;

type z = {
  a: string;
  b: number;
};
type x = z & Record<TempLiteralType, string>;
const a: x = {
  a: '1',
  b: 1,
  foo1: '2',
};
```
typebox ❌
```typescript
const TempLiteral = Type.TemplateLiteral('foo${number}');
type TempLiteralType = Static<typeof TempLiteral>;
const BrokenSchema = {
  body: Type.Intersect([
    Type.Object({
     a: Type.String(),
     b: Type.Number()
    }),
    Type.Record(TempLiteral, Type.String()),
  ]),
};
type staticBody = Static<typeof BrokenSchema.body>;
const x: staticBody  = {
  a: 'a',
  b: 1,
  MediaContentType0: 'a',
};
```
> Type '{ a: string; b: number; foo0: string; }' is not assignable to type 'Record<string, string>'.
    Property 'b' is incompatible with index signature.
      Type 'number' is not assignable to type 'string'.

**Comment by sinclairzx81** [comment]
@cwolter97 Hiya,

This is a good catch. Have published an update on `0.31.16` that should resolve the issue. The following should work on this version (note, renamed your `MediaContentType0` to `foo0`) as given by your TemplateLiteral pattern.

[TypeScript Link Here](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgFQJ5gKYBo4GUYCGMwAxnAL5wBmUEIcA5AAIDOwAdiQDYHBQD0MdBgBGEAB4MAsAChZJCOxbwAMnAC8KYQDpkGcDxgYVwI1AJcAFAyoQIAEgTsAriBEYo5BgEoA3PMVlPA1EWTg4MQATVAAuLUxtAEl2MxYMEhhLAG0w8PiMbQB5EQArdMyEXLyCOLQE-CgOAHNLbywq8JFanQA5V3coVo7yNo66goAldOhIyxUcce0G5tbRmXCAXTXyfxkhTDhlIlIAIQhokPxjkgAefYwIKjxtKNQAPl2FJXhxOKPiEhnC4hSrrOA1RgEBjtMFdOAARhh4VsEAADHEGFC4Px+HAIABrWQ7WRAA)

```typescript
import { Type, Static } from '@sinclair/typebox'

const L = Type.TemplateLiteral('foo${number}');
const S = {
  body: Type.Intersect([
    Type.Object({
     a: Type.String(),
     b: Type.Number()
    }),
    Type.Record(L, Type.String()),
  ]),
};
type staticBody = Static<typeof S.body>;
const x: staticBody  = {
  a: 'a',
  b: 1,
  foo0: 'a' // ok
};
```
Let me know if this solves the problem yourside.
Cheers
S

**Comment by sinclairzx81** [comment]
@cwolter97 Heya,

Going to close off this issue as things should be resolved on 0.31.16; but if you run into any issues, feel free to ping on this thread or just open a new issue.

Cheers
S
